tests/test_model.py

```diff
@@ -0,0 +1,189 @@
+        with open(tmp_path.joinpath(model.weights_file_name)) as f:
+            file_contents = f.read()
+        assert file_contents == weights_as_json
+
+    finally:
+        hf_api.delete_file(model.config_name, HF_WRITE_PATH)
+        hf_api.delete_file(model.weights_file_name, HF_WRITE_PATH)
+
+
+@pytest.mark.skipif(not hf_has_write_access, reason=HF_NO_ACCESS_MSG)
+def test_save_pretrained_push_to_hub_no_repo_id(
+    model, caplog, tmp_path, config_as_json, weights_as_json
+):
+    #   If no repo_id was provided, save_directory will be used instead.
+    # We need to name folder exactly as the Repository Name.
+    #   Token used for this tests should be issued by Repo owner, not anyone else with access,
+    # since repo will be looked up/created under current User's name.
+    #   In this case path also contains Username, which is not necessary, it will be ignored;
+    # But this is a simple solution to make sure folder has the right name if we change repos.
+    path = tmp_path.joinpath(HF_WRITE_PATH)
+    path.mkdir(parents=True, exist_ok=True)
+
+    try:
+        model.save_pretrained(save_directory=path, push_to_hub=True)
+
+        assert path.joinpath(model.config_name).exists()
+        with open(path.joinpath(model.config_name)) as f:
+            file_contents = f.read()
+        assert file_contents == config_as_json
+
+        assert path.joinpath(model.weights_file_name).exists()
+        with open(path.joinpath(model.weights_file_name)) as f:
+            file_contents = f.read()
+        assert file_contents == weights_as_json
+
+    finally:
+        hf_api.delete_file(model.config_name, HF_WRITE_PATH)
+        hf_api.delete_file(model.weights_file_name, HF_WRITE_PATH)
+
+
+@pytest.mark.skipif(not hf_has_write_access, reason=HF_NO_ACCESS_MSG)
+def test_push_to_hub(model):
+    try:
+        model.push_to_hub(HF_WRITE_PATH)
+        pretrained = TestModel.from_pretrained(HF_WRITE_PATH)
+        assert pretrained.is_from_pretrained
+        assert pretrained.config == model.config
+        assert pretrained.param == model.param
+
+    finally:
+        hf_api.delete_file(model.config_name, HF_WRITE_PATH)
+        hf_api.delete_file(model.weights_file_name, HF_WRITE_PATH)
+
+
+@pytest.mark.parametrize("config_path", [CONFIG_PATH, HF_PATH])
+def test_from_pretrained(model, config_path):
+    pretrained = TestModel.from_pretrained(config_path)
+    assert pretrained.is_from_pretrained
+    assert pretrained.config == model.config
+    assert pretrained.param == model.param
+
+
+def test_save_and_from_pretrained(model, tmp_path) -> None:
+    model.save_pretrained(tmp_path)
+    test_model = TestModel.from_pretrained(tmp_path)
+    assert test_model.config == model.config
+    assert test_model.param == model.param
+
+
+def test_auto_model_from_pretrained(model) -> None:
+    test_model = AutoModel.from_pretrained(CONFIG_PATH)
+    assert isinstance(test_model, TestModel)
```

Comment:
use exact check here:
```suggestion
    assert type(test_model) is TestModel
```

Reply:
> use exact check here

Should i do this only for Auto tests? I'll also change it for AutoTaskModule.

Reply:
for me it was most important for the Auto-case since its result needs to be the exact class defined in the type key of the config